src/compositemark/boxplot.ts

```diff
@@ -289,7 +289,7 @@ export function normalizeBoxPlot(
 
   const outlierLayersMixins = partLayerMixins<BoxPlotPartsMixins>(markDef, 'outliers', config.boxplot, {
     transform: [{filter: `(${fieldExpr} < ${lowerWhiskerExpr}) || (${fieldExpr} > ${upperWhiskerExpr})`}],
-    mark: 'point',
+    mark: {type: 'point', tooltip: {content: 'data'}},
```

Comment:
While I think this is useful, I think this makes the tool behave inconsistently. 

Personally I think tooltip is useful to be turned on by default in general, but I think we previously turned it off since it's distracting for some use cases.

Since we don't turn it on by default elsewhere, it's weird that this is the only place we turn on by default. 

Instead of turning tooltip on always, we should read boxplot's tooltip property and turn it on based on the specified tooltip property?

Reply:
Hello @kanitw , just a ping on my latest comment here, what do you think?

Reply:
I think it's important that we respect the tooltip property (which is default to false), so I'd prefer that we read the tooltip property.

Reply:
It turns out we have a default tooltip for box plots already, which as @kanitw said should not be the case. Let's fix this behavior and then also add tooltips for the outliers.

Reply:
I looked a bit more at this today and it seems like all composite marks in Vega-Lite currently have the tooltip on by default, see for example:

- https://vega.github.io/vega-lite/examples/layer_point_errorbar_ci.html
- https://vega.github.io/vega-lite/examples/layer_line_errorband_ci.html

I personally find that useful and it makes sense to me that boxplots also have a default tooltip since it is a composite mark. I can also see an argument for always having the tooltip off by default for all composite marks (although I personally find it more useful to have them on as they are currently), but I also think it makes sense to treat these differently from regular marks and there is an argument to be made that tooltips are more needed for complex marks. In either case, a change removing all tooltips from all composite marks seems to go beyond the scope of what I am trying to fix in this PR (which is consistency within the boxplot itself and making sure the box and outlier points are treated the same).